village/src/components/SmurfForm.js

```diff
@@ -1,57 +1,107 @@
-import React, { Component } from 'react';
+'use strict'
+
+/**
+ * Dependencies
+ */
+
+const React = require('react')
+const react_router_dom = require('react-router-dom')
+const styles = require('./styles/index')
+const Navbar = require('./Navbar')
+
+/**
+ * Constants
+ */
+
+const Component = React.Component
+const Link = react_router_dom.Link
+
+/**
+ * Define component
+ */
 
 class SmurfForm extends Component {
   constructor(props) {
-    super(props);
+    super(props)
     this.state = {
       name: '',
       age: '',
       height: ''
-    };
+    }
   }
 
   addSmurf = event => {
-    event.preventDefault();
-    // add code to create the smurf using the api
+    event.preventDefault()
+
+    this.props.addSmurf({
+      name: this.state.name,
+      age: this.state.age,
+      height: this.state.height
```

Comment:
I've found it easier to just send the whole state when using forms instead of typing out a whole new object. You could do something like 
```
this.props.addSmurf(this.state)
```